### PR TITLE
Angular: support getLegacyAngularInjector() even when angular is disabled.

### PR DESCRIPTION
**Pull request by ryantxu**
In angular, many internal services were exported in the injector context.  This meant we had many things usable by plugins that were not in the officially supported list.

We are now moving to deprecate/remove angular -- but do not yet have answers for better usage of these services.  In particular, we so not have a long term commitment/strategy for exposing the `dashboardSrv`

The only known plugin that uses this approach is https://github.com/grafana/grafana-iot-twinmaker-app

This approach lets us continue to kick the can on these internal/advanced features without breaking things when angular support is disabled.

NOTE: with these changes, twinmaker now works when angular is disabled

### CODE REVIEW

File: public/app/app.ts [hunk 1]
```diff
@@ -127,6 +128,7 @@ export class GrafanaApp {
       setPanelDataErrorView(PanelDataErrorView);
       setLocationSrv(locationService);
       setTimeZoneResolver(() => config.bootData.user.timezone);
+      initGrafanaLive();
```
Comment (ryantxu) [inline public/app/app.ts:131]: moved this out of the `GrafanaCtrl` so it is initialized regardless of the angular status!

**Review by torkelo — APPROVED** [review]

**Review by leventebalogh — APPROVED** [review]
LGTM, the only thing I think we should probably expose as well (temporarily) is the `timeSrv`.

**Review by iwysiu — APPROVED** [review]
I tested it with twinmaker, and it worked!

File: public/app/angular/loadAndInitAngularIfEnabled.ts [hunk 2]
```diff
@@ -18,5 +24,50 @@ export async function loadAndInitAngularIfEnabled() {
...
+            // we do not yet have a public interface for this
+            deprecationWarning('getLegacyAngularInjector', 'linkSrv');
+            return getLinkSrv();
+          }
+
+          case 'validationSrv': {
+            // we do not yet have a public interface for this
+            deprecationWarning('getLegacyAngularInjector', 'validationSrv');
+            return validationSrv;
+          }
```
[the remark this answers is not in the document]
Comment (ryantxu) [reply public/app/angular/loadAndInitAngularIfEnabled.ts:75]: thanks!  I did not see the ones registered in AngularApp!

**Pull request merged**